R/spaces.R

```diff
@@ -165,11 +165,276 @@ space_create <- function(name, spaces_key = NULL, spaces_secret = NULL, ...) {
+                     space,
+                     check_region = FALSE,
+                     key = spaces_key,
+                     secret = spaces_secret,
+                     base_url = spaces_base,
+                     ...)
+}
+
+#' Get information about an Object
+#' @param object (character) The Object to get information about
+#' @param space (character) The Space the Object is found in
+#' @param spaces_key (character) String containing a spaces access key. If
+#'   missing, defaults to value stored in an environment variable
+#'   \code{DO_SPACES_ACCESS_KEY}.
+#' @param spaces_secret (character) String containing the secret associated
+#'   with the spaces key. If missing, defaults to value stored in an environment
+#'   variable \code{DO_SPACES_SECRET_KEY}.
+#' @param ... Additional argument passed to \code{\link[aws.s3]{head_object}}
+#'
+#' @return A list of headers associated with the Object
+#' @export
+#'
+#' @references \url{https://developers.digitalocean.com/documentation/spaces/#get-object-info}
+#'
+#' @examples
+#' \dontrun{
+#' spaces_put_object("./some-file.txt", "my-object", "my-space")
+#' spaces_head_object("my-object", "my-space")
+#' }
+spaces_head_object <- function(object, space, spaces_key = NULL, spaces_secret = NULL, ...) {
+  spaces_key <- check_space_access(spaces_key)
+  spaces_secret <- check_space_secret(spaces_secret)
+
+  aws.s3::head_object(object,
+                      space,
+                      check_region = FALSE,
+                      key = spaces_key,
+                      secret = spaces_secret,
+                      base_url = spaces_base,
+                      ...)
+
+}
+
+#' Upload an Object to a Space
+#'
+#' @param file (character) The path to the file to upload.
+#' @param object (character) The key for the new Object. Defaults to the basename of the \code{path} returned by \code{\link{basename}}.
+#' @param space (character) The Space to upload the Object to
+#' @param spaces_key (character) String containing a spaces access key. If
+#'   missing, defaults to value stored in an environment variable
+#'   \code{DO_SPACES_ACCESS_KEY}.
+#' @param spaces_secret (character) String containing the secret associated
+#'   with the spaces key. If missing, defaults to value stored in an environment
+#'   variable \code{DO_SPACES_SECRET_KEY}.
+#' @param ... Additional argument passed to \code{\link[aws.s3]{put_object}}
+#'
+#' @return TRUE of FALSE dependong on whether the Object was uploaded
+#' @export
+#'
+#' @references \url{https://developers.digitalocean.com/documentation/spaces/#object}
+#'
+#' @examples
+#' \dontrun{
+#' spaces_put_object("./some-file.txt", "my-object", "my-space")
+#'
+#' # Let the function choose an Object key automatically based on the file's
+#' # basename
+#' spaces_put_object("./some-file.txt", space = "my-space")
+#' }
+spaces_put_object <- function(file, object = basename(file), space, spaces_key = NULL, spaces_secret = NULL, ...) {
```

Comment:
this function doesn't error well if the file doesn't exist - maybe an internal check for that?

Reply:
Roger.